### PR TITLE
Speed up DNSCacheManagerSpec + simplify some code in DNSCacheManager.

### PR DESCRIPTION
**Pull request by ham1**
Simplified and sped up DNSCacheManagerSpec

### CODE REVIEW

**Comment by vlsi** [comment]
Can you please clarify why do you mean this speeds up the test?

**Comment by ham1** [comment]
It only does a DNS test/lookup once rather than on each @IgnoreIf, I've also used @Requires as it's more readable.

So if you were running this with slow DNS it would timeout once rather than 4 times and save you 1 or many seconds.

**Comment by vlsi** [comment]
Ah, thanks. The change looks good to me.

@pmouawad , any comments?

File: src/protocol/http/src/test/groovy/org/apache/jmeter/protocol/http/control/DNSCacheManagerSpec.groovy [hunk 1]
```diff
@@ -17,23 +17,26 @@
...
 
-    private static final String[] VALID_DNS_SERVERS  = ResolverConfig.getCurrentConfig().servers()
+    private static final String[] VALID_DNS_SERVERS = ResolverConfig.getCurrentConfig().servers()
     private static final String INVALID_DNS_SERVER = "512.1.1.1"
+    private static final String VALID_HOSTNAME = "jmeter.apache.org"
 
-    static def localDNSResolverFailed() {
+    private static final boolean localDNSResolverOK = {
         try {
+            println "localDNSResolverOK"
```
Comment (pmouawad) [inline src/protocol/http/src/test/groovy/org/apache/jmeter/protocol/http/control/DNSCacheManagerSpec.groovy:35]: should this remain ?

**Review by pmouawad — APPROVED** [review]

File: src/protocol/http/src/test/groovy/org/apache/jmeter/protocol/http/control/DNSCacheManagerSpec.groovy (hunk 1 above)
Comment (vlsi) [reply src/protocol/http/src/test/groovy/org/apache/jmeter/protocol/http/control/DNSCacheManagerSpec.groovy:35]: Nice catch. @ham1, would you please remove that?
```suggestion
```

**Pull request merged**